Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,148 @@ This document includes four main sections:
       PEP written by Marc-André Lemburg.
 
 
+.. We use the following practises for SQL code:
+   - UPPERCASE for keywords
+   - SnakeCase for schema
+   - single quotes for string literals
+   - singular for table names
+   - if needed, use double quotes for table and column names
+
 .. _sqlite3-tutorial:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating an on-disk movie database :file:`tutorial.db`
+an filling it with meta-data about Monty Python movies.
+A fundamental understanding of database concepts,
+like `transactions`_ and `cursors`_, is assumed.
+
+To start working with an SQLite database,
+you must first open a connection to it.
+Call the :func:`sqlite3.connect` function
+to create a connection to the database :file:`tutorial.db`::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+SQLite will implicitly create an empty database file :file:`tutorial.db`
+in the current working directory if it does not exist.
```

Comment:
This still seems a bit backwards to me, since it says in order that:
1. we need to connect to the db (which db?)
2. we can use `sqlite.connect` to connect (connect to what?)
3. we don't need to create the db since it will create it automatically

I think it would be better to say:
1. we have to create a new empty db and connect to it
2. (why connect?) the connection will allow us to work with it
3. (how to create/connect?) `sqlite.connect` connects to the db and creates it if it doesn't exist

Reply:
I've reworded it so it now says:

- we have to create a new and empty db
- we need to connect to it, because we need the connections to work with the db
- show the user how to connect
- explain that, as a result of the connect, SQLite will implicitly create the file on-disk and a connection object is now available